src/test/java/com/emc/object/s3/S3JerseyClientTest.java

```diff
@@ -1107,6 +1116,92 @@ public void testPutObjectPreconditions() {
         client.putObject(request);
     }
 
+    @Test
+    public void testDeleteObjectPreconditions() {
+        Assume.assumeTrue("ECS version must be at least 3.7", ecsVersion != null && ecsVersion.compareTo("3.7") >= 0);
+        String key = "testDeletePreconditions";
+        String content = "hello Delete preconditions!";
+
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.MINUTE, -5); // 5 minutes ago
+
+        client.putObject(getTestBucket(), key, content, "text/plain");
+        String etag = client.getObjectMetadata(getTestBucket(), key).getETag();
+        String etag2 = "d41d8cd98f00b204e9800998ecf8427e"; //non-matching Etag
+
+        DeleteObjectRequest request = new DeleteObjectRequest(getTestBucket(), key);
+
+        // test if-unmodified fail
+        request.withIfUnmodifiedSince(cal.getTime());
+        try {
+            client.deleteObject(request);
+            Assert.fail("expected 412");
+        } catch (S3Exception e) {
+            Assert.assertEquals(412, e.getHttpCode());
+        }
+
+        // test if-unmodified and if-match(correct etag) fail
+        request.withIfUnmodifiedSince(cal.getTime()).withIfMatch(etag);
+        try {
+            client.deleteObject(request);
+            Assert.fail("expected 412");
+        } catch (S3Exception e) {
+            Assert.assertEquals(412, e.getHttpCode());
+        }
+
+        // test if-unmodified pass
+        cal.add(Calendar.MINUTE, 10); // 5 minutes from now
+        request.withIfUnmodifiedSince(cal.getTime());
+        client.deleteObject(request);
+
+
+        client.putObject(getTestBucket(), key, content, "text/plain");
+        // test if-unmodified and if-match(non-matching etag) fail
+        request.withIfUnmodifiedSince(null).withIfMatch(etag2);
+        try {
+            client.deleteObject(request);
+            Assert.fail("expected 412");
+        } catch (S3Exception e) {
+            Assert.assertEquals(412, e.getHttpCode());
+        }
+
+        // test if-match(non-matching etag) fail
+        request.withIfUnmodifiedSince(null).withIfMatch(etag2);
+        try {
+            client.deleteObject(request);
+            Assert.fail("expected 412");
+        } catch (S3Exception e) {
+            Assert.assertEquals(412, e.getHttpCode());
+        }
+
+        //test if-match pass
+        request.withIfUnmodifiedSince(null).withIfMatch(etag);
+        client.deleteObject(request);
+
+        //test if-match pass with deleted key
+        request.withIfUnmodifiedSince(null).withIfMatch(etag);
+        client.deleteObject(request);
+
+        client.putObject(getTestBucket(), key, content, "text/plain");
+        // test if-match * (key exists) pass
+        request.withIfMatch("*");
+        client.deleteObject(request);
+
+        // test if-match * pass with deleted key
+        request.withIfMatch("*");
```

Comment:
is the expectation here that all of these will pass only because the object does not exist?

Reply:
yes, pre-condition is expected to pass on non-existing keys. Test cases refined.